hazelcast/src/test/java/com/hazelcast/journal/AbstractEventJournalBasicTest.java

```diff
@@ -485,34 +484,34 @@ public void allowReadingWithFutureSeq() throws Exception {
         assertEquals(-1, state.getNewestSequence());
         assertEventJournalSize(context.dataAdapter, 0);
 
-        final String key = randomPartitionKey();
         final Integer value = RANDOM.nextInt();
         final CountDownLatch latch = new CountDownLatch(1);
         final int startSequence = 1;
 
         final BiConsumer<ReadResultSet<EJ_TYPE>, Throwable> callback = (response, t) -> {
             if (t == null) {
+                latch.countDown();
                 assertEquals(1, response.size());
                 final EventJournalEventAdapter<String, Integer, EJ_TYPE> journalAdapter = context.eventJournalAdapter;
                 final EJ_TYPE e = response.get(0);
 
                 assertEquals(ADDED, journalAdapter.getType(e));
-                assertEquals(key, journalAdapter.getKey(e));
                 assertEquals(value, journalAdapter.getNewValue(e));
-                assertNotEquals(startSequence + response.readCount(), response.getNextSequenceToReadFrom());
-                assertEquals(1, response.getNextSequenceToReadFrom());
-                latch.countDown();
             } else {
-                t.printStackTrace();
+                rethrow(t);
```

Comment:
If the event journal read request was completed exceptionally, rethrow the exception